pocketbooksync.koplugin/main.lua

```diff
@@ -85,13 +85,21 @@ function PocketbookSync:doSync(data)
     end
     local book_id = row[1]
 
+	local h1 = io.popen('test=$(readlink -f /mnt/ext1/system/profiles/.current); if [ $? != 0 ]; then echo "default"; else echo ${test##*/}; fi')
+	local h2 = h1:read('*a')
+	local profile_name = h2:gsub('[\n\r]', '')
+	h1:close()
```

Comment:
I wonder if we can just use this:
https://github.com/koreader/koreader-base/blob/8f5f38d732bba170abdae5df015f9f4b475fac6e/ffi/inkview_h.lua#L1324

Running a bash script on every page turn seems quite excessive. If the above doesn't work, we can still perhaps just get the profile name once? I've never used profiles on my PocketBook but I somehow doubt it makes sense to change them while keeping KOReader running…

Reply:
I w3as thinking the same thing the other day.  I've amended the code and PR accordingly.

I played with the inkview function `GetCurrentProfile()` but all I could get it to output was `cdata<char *>: 0xa6a9fe68`

Reply:
Got it...

needed to be:  `local test1234 = ffi.string(inkview.GetCurrentProfile())`

Reply:
Cool, I'll test it on mine (InkPad 3). 👍

Which one do you have BTW?

Reply:
Okay, almost works. The original version of this PR dealt with there being no profiles but that got lost somewhere, so I had to adjust it to handle `inkview.GetCurrentProfile()` returning NULL.

Here's my adjustments: https://github.com/liskin/pocketbooksync.koplugin/commit/9c509779f8626a930e5e6badbb0a921fff27f5e5

I refactored the code slightly as well.

Reply:
Looks good!  Just line 22 needs changing (i made a comment on it).   Makes sense to only query the database if needed.

I'm using a PB632, aka HD3.

Reply:
> P.s. sorry to be cheeky... but you seem like you know what you're doing with LUA! I'm stuck modding another koreader plugin. Another user has offered a fix, but I'm lost in implementing it. If you have time could you take a peek? [koreader/koreader#11591](https://github.com/koreader/koreader/discussions/11591) (If you like crosswords and don't already know about this, it's an awesome plugin! 😄 )

"Seem" is correct, in reality I spend more time swearing at the Lua documentation being shit than actually getting stuff done. 🤣
I'm not at all familiar with the UI widgets stuff in KOReader so I'd probably have to invest a couple hours into this and I'm afraid it's not quite that high a priority right now. Apologies! 😳

Reply:
No worries, it was a big ask!

You're so right about the LUA docs.  Many curse words used and there will be many more!  😆